packages/langium/src/lsp/language-server.ts

```diff
@@ -519,6 +526,47 @@ export function createCallHierarchyRequestHandler<P extends CallHierarchyIncomin
     };
 }
 
+export function addTypeHierarchyHandler(connection: Connection, sharedServices: LangiumSharedServices): void {
+    connection.languages.typeHierarchy.onPrepare(
```

Comment:
Here we could check whether there is a language with a type hierarchy provider before we register the handlers.